test/cluster/security_test.go

```diff
@@ -0,0 +1,435 @@
+	require.NotNil(t, rootCAPrivKey)
+
+	keyPath, certPath, rootCACertPath, err := server5.CreateNewCryptoMaterials(certRootCA, rootCAPrivKey)
+	localConfig := &config.LocalConfiguration{
+		Replication: config.ReplicationConf{
+			TLS: config.TLSConf{
+				Enabled:               true,
+				ServerCertificatePath: certPath,
+				ServerKeyPath:         keyPath,
+				ClientCertificatePath: certPath,
+				ClientKeyPath:         keyPath,
+				CaConfig: config.CAConfiguration{
+					RootCACertsPath: []string{rootCACertPath},
+				},
+			},
+		},
+	}
+	require.NoError(t, server5.CreateConfigFile(localConfig))
+
+	configEnv, err = c.Servers[leaderIndex].QueryConfig(t, "admin")
+	require.NoError(t, err)
+	require.NotNil(t, configEnv)
+	newConfig := configEnv.GetResponse().GetConfig()
+	newConfig.ConsensusConfig.Members = append(newConfig.ConsensusConfig.Members, peer5)
+	newConfig.Nodes = append(newConfig.Nodes, node5)
+
+	// add server with config-tx
+	txID, rcpt, err := c.Servers[leaderIndex].SetConfigTx(t, newConfig, configEnv.GetResponse().GetMetadata().GetVersion(), c.Servers[leaderIndex].AdminSigner(), "admin")
+	require.NoError(t, err)
+	require.NotNil(t, rcpt)
+	require.True(t, txID != "")
+	require.True(t, len(rcpt.GetHeader().GetValidationInfo()) > 0)
+	require.Equal(t, types.Flag_VALID, rcpt.Header.ValidationInfo[rcpt.TxIndex].Flag)
+	t.Logf("tx submitted: %s, %+v", txID, rcpt)
+
+	configBlockResponseEnvelope, err := c.Servers[leaderIndex].QueryConfigBlockStatus(t)
+	require.NoError(t, err)
+	require.NotNil(t, configBlockResponseEnvelope)
+
+	err = ioutil.WriteFile(server5.BootstrapFilePath(), configBlockResponseEnvelope.Response.Block, 0666)
+	require.NoError(t, err)
+
+	config, err := config.Read(server5.ConfigFilePath())
+	require.NoError(t, err)
+	require.NotNil(t, config.LocalConfig)
+	require.Nil(t, config.SharedConfig)
+	require.NotNil(t, config.JoinBlock)
+
+	c.AddNewServerToCluster(server5)
+	require.EqualError(t, c.StartServer(server5), "failed to start the server: node-5")
+}
+
+// Scenario:
+// - start a 3 servers cluster with TLS enabled.
+// - change server-2 config file
+func TestTLSChangeNodeCA(t *testing.T) {
+	dir, err := ioutil.TempDir("", "int-test")
+	require.NoError(t, err)
+
+	nPort, pPort := getPorts(3)
+	setupConfig := &setup.Config{
+		NumberOfServers:     3,
+		TestDirAbsolutePath: dir,
+		BDBBinaryPath:       "../../bin/bdb",
+		CmdTimeout:          10 * time.Second,
+		BaseNodePort:        nPort,
+		BasePeerPort:        pPort,
+		ClusterTLSEnabled:   true,
+		CheckRedirectFunc: func(req *http.Request, via []*http.Request) error {
+			return errors.Errorf("Redirect blocked in test client: url: '%s', referrer: '%s', #via: %d", req.URL, req.Referer(), len(via))
+		},
+	}
+	c, err := setup.NewCluster(setupConfig)
+	require.NoError(t, err)
+	defer c.ShutdownAndCleanup()
+
+	//require.NoError(t, c.StartServer(c.Servers[0]))
+	//require.NoError(t, c.StartServer(c.Servers[1]))
+	require.NoError(t, c.Start())
```

Comment:
you are starting all the cluster, so wait for 0,1,2 to get a common leader, then shutdown 2 and continue, i.e. shutdown 2, wait for 0,1 leader, changing the crypto on 2.